.github/workflows/build-container.yml

```diff
@@ -3,16 +3,94 @@ name: Build CI Container
 on:
   repository_dispatch:
     types: [build-container]
+    inputs:
+      push_image:
+        description: If the image should be pushed to the registry
+        type: boolean
+        default: true
   workflow_dispatch:
-  push:
-    branches:
-      - "*"
-      - "!main"
+    inputs:
+      commit:
+        description: 'Initial commit to layer the resulting image'
+        required: true
+        type: string
+      push_image:
+        description: If the image should be pushed to the registry
+        type: boolean
+        default: true
 
+  # Build container on PR open and switch to ready
+  pull_request:
+    types: [opened, ready_for_review]
+ 
 jobs:
-  build-container:
+  parse-inputs:
     runs-on: ubuntu-latest
-    container: hashicorp/packer:1.11
+    outputs:
+      commits: ${{ steps.check.outputs.commits }}
+      action_build_container: ${{ steps.check.outputs.action_build_container }}
+      base_commit: ${{ steps.check.outputs.base_commit }}
+      last_commit: ${{ steps.check.outputs.last_commit }}
+      version: ${{ steps.check.outputs.version }}
+      pipeline_java_commit: ${{ steps.check.outputs.pipeline_java_commit }}
+      pipeline_java_version: ${{ steps.check.outputs.pipeline_java_version }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Check and parse inputs
+        id: check
+        run: |
+          # Check whether the commit was provided
+          if [ "z${{ inputs.commit }}" != "z" ]; then
+
+            # We're on some commit, so rebuild the whole image
+            ## find all commits to layer in the image
+            echo "commits<<EOF" >> $GITHUB_OUTPUT
```

Comment:
todo: Update comment for why `commit` doesn't have `=` and add comment for why version is not here